js/bruh-dash.js

```diff
@@ -1,83 +1,500 @@
 var global = window || GLOBAL;
 
 global.bruhdash = {
-  chunk: function(){
-
+  /**
+    * Creates an array of elements split into chunks
+    * of length size.
+    * If array can’t be split evenly,
+    * the final chunk will contain the remaining elements.
+    * @param array {Array} The array to process
+    * @param size {Number} The length of each chunk
+    * @returns {Array} The new array of chunks
+    */
+  chunk: function(array, size){
+    if(size < 1 || isNaN(size)) {
+      size = 1;
+    }
+    var numChunks = Math.ceil(array.length / size);
+    var chunkedArray = [];
+    var chunkOfArray = [];
+
+    chunkOfArray = array.splice(0, size);
+    chunkedArray[0] = chunkOfArray;
+    for(var i = 1; i < numChunks; i++) {
+      chunkOfArray = [];
+      chunkOfArray = array.splice(0, size);
+      chunkedArray[i] = chunkOfArray;
+    }
+    return chunkedArray;
   },
-
-  compact: function() {
-
+  /**
+    * Creates an array with all falsey values removed.
+    * Falsey values: false, null, 0, "", undefined, and NaN.
+    * @param array {Array} the array to compact
+    * @returns {Array} the new array of filtered values
+    */
+  compact: function(array) {
+    var a;
+    var newArray = [];
+
+    for(var x = 0; x < array.length; x++) {
+      newArray[x] = array[x];
+    }
+    for(var i = 0; i < array.length; i++) {
+      a = array[i];
+        if(a === false || a === null || a === 0 || a === "" || a === undefined || isNaN(a)) {
```

Comment:
There's a shorter way to write this condition.

Leverage the truthiness value of `a`!
💁